// @particle-network/auth-core-modal/esm/selectSecurityAccount-AZ45XSDR.js
"use client";
import {
  displayEmail,
  displayPhone,
  particle_drawer_default,
  useAuthCoreModal,
  useCustomNavigate,
  useParticleAuth
} from "./chunk-BNR22FAP.js";
import "./chunk-LQ53OFQ3.js";

// src/pages/account/selectSecurityAccount/index.tsx
import { AuthType } from "@particle-network/auth-core";
import { Button } from "antd";
import React from "react";
import { useTranslation } from "react-i18next";

// src/pages/account/selectSecurityAccount/index.less
var selectSecurityAccount_default = ".select-account-drawer .ant-drawer-body {\n  padding: 18px !important;\n}\n.select-account {\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n  gap: 15px;\n}\n.select-account .close-icon {\n  position: absolute;\n  top: 10px;\n  right: 10px;\n}\n.select-account .account-btn {\n  width: 100%;\n  height: 47px;\n  border: none;\n  border-radius: var(--primary-btn-border-radius);\n  outline: none;\n  font-size: var(--primary-btn-font-size);\n  line-height: 22px;\n  color: var(--primary-btn-color);\n  background: var(--primary-btn-background-color);\n}\n.select-account .account-btn:hover {\n  opacity: var(--hover-opacity);\n}\n.select-account .select-account-title {\n  padding-top: 24px;\n  padding-bottom: 24px;\n  font-weight: 500;\n  font-size: 20px;\n  line-height: 1;\n  color: var(--text-color);\n}\n";

// src/pages/account/selectSecurityAccount/index.tsx
var SelectSecurityAccount = (props) => {
  var _a, _b, _c;
  const { t } = useTranslation();
  const { visible, state } = props;
  const { showSelectSecurityAccount, userInfo } = useParticleAuth();
  const navigate = useCustomNavigate();
  const { authCoreModal } = useAuthCoreModal();
  const navigateToVerify = (account, authType) => {
    if (state) {
      navigate("/account/verify", {
        state: {
          account,
          ...state
        }
      });
    } else {
      navigate("/account/set-password", {
        state: {
          account
        }
      });
    }
    showSelectSecurityAccount(false);
  };
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement("style", null, selectSecurityAccount_default), /* @__PURE__ */ React.createElement(
    particle_drawer_default,
    {
      title: (_a = t("account.modal_select_verification")) != null ? _a : "",
      placement: "bottom",
      closable: false,
      className: "select-account-drawer",
      visible,
      maskClosable: true,
      forceRender: true,
      height: 220,
      onClose: () => showSelectSecurityAccount(false),
      getContainer: () => {
        return authCoreModal.rootBody;
      }
    },
    /* @__PURE__ */ React.createElement("div", { className: "select-account" }, /* @__PURE__ */ React.createElement(
      Button,
      {
        className: "account-btn",
        onClick: () => {
          var _a2;
          return navigateToVerify((_a2 = userInfo == null ? void 0 : userInfo.security_account) == null ? void 0 : _a2.email, AuthType.email);
        }
      },
      `${t("account.email")} - ${displayEmail(((_b = userInfo == null ? void 0 : userInfo.security_account) == null ? void 0 : _b.email) || "")}`
    ), /* @__PURE__ */ React.createElement(
      Button,
      {
        className: "account-btn",
        onClick: () => {
          var _a2;
          return navigateToVerify((_a2 = userInfo == null ? void 0 : userInfo.security_account) == null ? void 0 : _a2.phone, AuthType.phone);
        }
      },
      `${t("account.mobile")} - ${displayPhone(((_c = userInfo == null ? void 0 : userInfo.security_account) == null ? void 0 : _c.phone) || "")}`
    ))
  ));
};
var selectSecurityAccount_default2 = SelectSecurityAccount;
export {
  selectSecurityAccount_default2 as default
};
